File: survey/src/main/java/com/fc/survey/entity/Survey.java
package com.fc.survey.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import java.util.List;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Comment;

@Entity
@Getter
@NoArgsConstructor
public class Survey extends BaseTimeEntity{

  @Comment("설문조사 테이블 식별자")
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Comment("설문조사 이름")
  @Column
  private String name;

  @Comment("설문조사 설명")
  @Column
  private String description;

  @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL)
  private List<Question> questions;

  @Builder
  public Survey(String name, String description, List<Question> questions) {
    this.name = name;
    this.description = description;
    this.questions = questions;
  }

  public void setQuestions(List<Question> questions) {
    this.questions = questions;
  }
}

File: survey-core/src/main/kotlin/org/icd/surveycore/domain/survey/Survey.kt
package org.icd.surveycore.domain.survey

import jakarta.persistence.*
import org.hibernate.annotations.Comment
import org.icd.surveycore.domain.support.BaseEntity
import org.icd.surveycore.domain.surveyItem.SurveyItem
import org.icd.surveycore.domain.surveyresponse.SurveyResponse

@Entity
@Table(name = "survey")
class Survey(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @Comment("설문조사 이름")
    val name: String,
    @Comment("설문조사 설명")
    val description: String? = null,
    @OneToMany(fetch = FetchType.EAGER, mappedBy = "survey", cascade = [CascadeType.ALL], orphanRemoval = true)
    val items: MutableList<SurveyItem> = mutableListOf(),
    @OneToMany(fetch = FetchType.EAGER, mappedBy = "survey", cascade = [CascadeType.ALL], orphanRemoval = true)
    val responses: MutableList<SurveyResponse> = mutableListOf(),
) : BaseEntity() {
    companion object {
        fun of(
            name: String,
            description: String?,
        ): Survey {
            return Survey(
                name = name,
                description = description
            )
        }
    }

    fun addItem(item: SurveyItem) {
        this.items.add(item)
    }

    fun getActiveItems(): List<SurveyItem> {
        return items.filter { it.isActive }
    }
}